cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -369,17 +334,22 @@ func (m *asgCache) regenerate() error {
 	newAsgToInstancesCache := make(map[AwsRef][]AwsInstanceRef)
 
 	// Build list of known ASG names
-	refreshNames, err := m.buildAsgNames()
+	configuredAsgNames, err := m.getConfiguredAsgNames()
 	if err != nil {
 		return err
 	}
 
 	// Fetch details of all ASGs
-	klog.V(4).Infof("Regenerating instance to ASG map for ASGs: %v", refreshNames)
-	groups, err := m.awsService.getAutoscalingGroupsByNames(refreshNames)
+	namedGroups, err := m.awsService.getAutoScalingGroupsByNames(configuredAsgNames)
 	if err != nil {
 		return err
 	}
+	taggedGroups, err := m.awsService.getAutoScalingGroupsByTags(m.asgAutoDiscoverySpecs)
+	if err != nil {
+		return err
+	}
+
+	groups := append(namedGroups, taggedGroups...)
```

Comment:
is there a chance for duplicate asgs being in `groups` now?

Reply:
You're right; I'll update to add back de-duping checks.